### PR TITLE
RR/milestone-01-code-refactoring: removed deprecated code and refactor setting related files

### PR DESCRIPTION
**Pull request by rupalirajput2597**
## What does this PR do?

1. Removes deprecated code
2. Refactors Setting related screen with svelte 5
3. Setting screen deep linking implementation

### Related Issues & PRs

- Jonah's slack message : https://baragaun.slack.com/archives/C084NTKGT26/p1741819836233139

#### Key Decisions

- **Implemented Approach**:
  
- **Alternatives Considered**:
  
- **Rationale**:
  

## Breaking Changes

- [ ] This PR includes breaking changes
  
No breaking changes so far

### Limitations & Concerns

- **Known limitations**:
1. Test cases are missing
2. data layer Integration is pending
- **Edge cases**:
- **Potential risks**:

---

### UI Changes

- [x] This PR includes UI changes

### Screenshots
| Description | Screenshot |
|-------------|------------|
| Before      |  
Screenshot 2025-03-13 at 10 08 43 PM

| After       | 
Screenshot 2025-03-13 at 10 25 18 PM

---

### Reviewers

- [x] Reviewer 1: @jonahtwalker 
- [x] Reviewer 2: @shadowbrush 
- [x] Reviewer 3: @jahanvi-baragaun 
- [x] Reviewer 4: @RaghvindYadav 
- [x] Reviewer 5: @evanschimpf 
- [x] Reviewer 6: @rashmit530 
- [x] Reviewer 7: @Copilot

### Next Steps

- [x] Follow-up tasks are created
- Just one task is created in issues there will be more integration tasks
  - [x] https://github.com/baragaun/first-spark-app/issues/43

### How to Test

1. Prerequisites:
  
4. Test steps:
   
5. Expected results:
   

## Final Checklist
- [x] Linked related issues
- [x] Rebased with target branch
- [x] No lint errors
- [x] QA'd in multiple browsers
- [ ] Added/updated tests (if applicable)
- [x] Mobile responsive (if applicable)
- [x] Updated documentation (if applicable)
- [x] Notified the Team of this updates readiness

---

## AI Usage

- [x] AI tools were used in development
  - **Tool used**:
    
  VS code extension - Augment code
  - **Prompts used**:
    
    - **Initial prompt**: 

-     Fix the selected code. The IDE reports the following issues on L11: 'page' is deprecated.
-     I wants to show http://localhost:5173/settings/account and http://localhost:5173/settings/notification when user. selects account and notification tab from settings screen, right now we are just showing http://localhost:5173/settings url. can you please do these changes without breaking existing functionality. please do follow svelte-complete-medium.txt file before doing any changes. I want pure svelte 5 code
-     I hope you have followed the svelte-complete-medium for svelte 5 changes in setting files
-     I wan you to analyse @src/lib/components/settings-ui/notification-settings.svelte file as per svelte-complete-medium.txt and remember do not remove any commented code from the file as I have kept it for future use
-     I wan you to analyse @src/lib/components/settings-ui/account-settings.svelte file as per svelte-complete-medium.txt. and update if any non svelte 5 code and remember do not remove any commented code from the file I have kept it for future use
-     dialog-content.svelte file as per svelte-complete-medium.txt. and update if any non svelte 5 code
-     dialog.svelte file as per svelte-complete-medium.txt. and update if any non svelte 5 code
-     some more to update remaining dialog files

- [ ] No AI tools were used

## References

- **Documentation**:
- **Discussions**:
- **Related PRs**:

### CODE REVIEW

File: src/routes/faq/+page.svelte [hunk 1]
```diff
@@ -6,21 +6,25 @@
...
+    faqSections[sectionIndex].items[itemIndex].isOpen =
+      !faqSections[sectionIndex].items[itemIndex].isOpen;
   }
 
-  $effect(() => {
-    if (typeof window !== 'undefined' && getOpenQuestionsCount() > 0) {
-      console.log(`FAQ sections open: ${getOpenQuestionsCount()}`);
+  $effect.root(() => {
+    if (typeof window !== 'undefined' && openQuestionsCount > 0) {
+      console.log(`FAQ sections open: ${openQuestionsCount}`);
```
Comment (jonahtwalker) [inline src/routes/faq/+page.svelte]: Do we need this log?

File: src/routes/settings/+layout.svelte [hunk 2]
```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+  import { cn } from '$lib/utils';
+  import { Button } from '$lib/components/ui/button';
+  import { getStores } from '$app/stores';
+  import { goto } from '$app/navigation';
+
+  const { page } = getStores();
```
Comment (jonahtwalker) [inline src/routes/settings/+layout.svelte]: Why aren't you using `state`?

File: src/lib/components/app-sidebar.svelte [hunk 3]
```diff
@@ -8,56 +8,41 @@
   import { useSidebar } from '$lib/components/ui/sidebar/context.svelte.js';
   import { fly } from 'svelte/transition';
   import { quartOut } from 'svelte/easing';
-  import { page } from '$app/stores';
+  import { getStores } from '$app/stores';
```
Comment (jonahtwalker) [inline src/lib/components/app-sidebar.svelte]: Why do you use Stores instead of State?

File: src/lib/components/app-sidebar.svelte [hunk 4]
```diff
@@ -8,56 +8,41 @@
...
-
-  // Check if item is active
-  function isActive(url: string): boolean {
-    return currentPath === url;
-  }
+  // Derived values using runes
+  let currentPath = $derived.by(() => $page.url.pathname);
+  let sidebarState = $derived.by(() => sidebar.state);
+  let itemsList = $derived.by(() => Object.values(items));
+  let isActive = (url: string) => {
```
Comment (jonahtwalker) [inline src/lib/components/app-sidebar.svelte]: The component from shad already handles this, see [this line in my pr](https://github.com/baragaun/first-spark-app/pull/44/files#:~:text=isActive%3D%7BactiveItem.title%20%3D%3D%20item.title%7D)

File: src/svelte-coding-instructions.txt
Comment (jonahtwalker) [inline src/svelte-coding-instructions.txt:1]: Is this being removed in favor of the medium Svelte docs file?

**Review by jonahtwalker — CHANGES_REQUESTED** [review]
Thanks Rupali. Did you look at [the component docs](https://next.shadcn-svelte.com/docs/components/dialog) before making these updates?

If we've added Svelte4 components from shad via `shadcn-svelte@latest` instead of using the Svelte5 versions via `shadcn-svelte@next`, we should just replace the components entirely, and not further customize them ourselves. Take a look at [`dialog-description.svelte`](https://github.com/huntabyte/shadcn-svelte/blob/main/sites/docs/src/lib/registry/default/ui/dialog/dialog-description.svelte), for instance.

I am going to QA and I'll get back with additional feedback. Thanks again!

File: src/svelte-coding-instructions.txt
Comment (rupalirajput2597) [reply src/svelte-coding-instructions.txt:1]: I have removed this, because It is already there in `docs` folder, so just removed extra one

File: src/lib/components/app-sidebar.svelte (hunk 4 above)
Comment (rupalirajput2597) [reply src/lib/components/app-sidebar.svelte]: ohh, I got it. Let me update my code the same way

File: src/routes/settings/+layout.svelte (hunk 2 above)
Comment (rupalirajput2597) [reply src/routes/settings/+layout.svelte]: I initially asked the AI to remove deprecated changes, but I didn't realize it wasn't Svelte 5 code. After reviewing the [migration guide for app-store](https://svelte.dev/docs/kit/migrating-to-sveltekit-2#SvelteKit-2.12:-$app-stores-deprecated), I now understand the necessary updates and will make the corrections accordingly. Thanks for catching that, Jonah!

File: src/routes/faq/+page.svelte (hunk 1 above)
Comment (rupalirajput2597) [reply src/routes/faq/+page.svelte]: It is not needed. will remove it!

File: src/lib/components/app-sidebar.svelte (hunk 3 above)
Comment (rupalirajput2597) [reply src/lib/components/app-sidebar.svelte]: Same here, will make the corrections according to [migration guide for app-store](https://svelte.dev/docs/kit/migrating-to-sveltekit-2#SvelteKit-2.12:-$app-stores-deprecated). Thank you!

**Comment by rupalirajput2597** [comment]
> Thanks Rupali. Did you look at [the component docs](https://next.shadcn-svelte.com/docs/components/dialog) before making these updates?
> 
> If we've added Svelte4 components from shad via `shadcn-svelte@latest` instead of using the Svelte5 versions via `shadcn-svelte@next`, we should just replace the components entirely, and not further customize them ourselves. Take a look at [`dialog-description.svelte`](https://github.com/huntabyte/shadcn-svelte/blob/main/sites/docs/src/lib/registry/default/ui/dialog/dialog-description.svelte), for instance.
> 
> I am going to QA and I'll get back with additional feedback. Thanks again!

Hello Jonah,

Thank you for reviewing my PR and providing valuable feedback. I have updated the Dialog component to the Svelte 5 version using shadcn-svelte@next and implemented the other suggested changes as well. I would appreciate it if you could review it again and let me know if you have any further feedback.

Looking forward to your thoughts!

**Comment by jonahtwalker** [comment]
Thank you Rupali! I've cherry picked some things here and incorporated them into my PR, #44 

Will you give that a review when you've some time? I've left comments on just about all of the files. I'd like to do a bit more in the menus and then merge that in as soon as we can so we can get to the node-client stuff this week.

Thanks again! 👍🏼

**Comment by jonahtwalker** [comment]
Updates from this PR have been combined with #44
